src/python/grpcio_tests/tests_aio/unit/interceptor_test.py

```diff
@@ -577,6 +577,106 @@ class Interceptor(aio.UnaryUnaryClientInterceptor):
 
             self.assertEqual(await call.code(), grpc.StatusCode.OK)
 
+    async def test_add_done_callback_before_finishes(self):
+        called = asyncio.Event()
+        interceptor_can_continue = asyncio.Event()
+
+        def callback(call):
+            called.set()
+
+        class Interceptor(aio.UnaryUnaryClientInterceptor):
+
+            async def intercept_unary_unary(self, continuation,
+                                            client_call_details, request):
+
+                await interceptor_can_continue.wait()
+                call = await continuation(client_call_details, request)
+                return call
+
+        async with aio.insecure_channel(self._server_target,
+                                        interceptors=[Interceptor()
+                                                     ]) as channel:
+
+            multicallable = channel.unary_unary(
+                '/grpc.testing.TestService/UnaryCall',
+                request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+                response_deserializer=messages_pb2.SimpleResponse.FromString)
+            call = multicallable(messages_pb2.SimpleRequest())
+            call.add_done_callback(callback)
+            interceptor_can_continue.set()
+            await call
+
+            try:
+                await asyncio.wait_for(called.wait(), timeout=0.1)
+            except:
+                self.fail("Callback was not called")
+
+    async def test_add_done_callback_after_finishes(self):
+        called = asyncio.Event()
+
+        def callback(call):
+            called.set()
+
+        class Interceptor(aio.UnaryUnaryClientInterceptor):
+
+            async def intercept_unary_unary(self, continuation,
+                                            client_call_details, request):
+
+                call = await continuation(client_call_details, request)
+                return call
+
+        async with aio.insecure_channel(self._server_target,
+                                        interceptors=[Interceptor()
+                                                     ]) as channel:
+
+            multicallable = channel.unary_unary(
+                '/grpc.testing.TestService/UnaryCall',
+                request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+                response_deserializer=messages_pb2.SimpleResponse.FromString)
+            call = multicallable(messages_pb2.SimpleRequest())
+
+            await call
+
+            call.add_done_callback(callback)
+
+            try:
+                await asyncio.wait_for(called.wait(), timeout=0.1)
```

Comment:
Let's use a constant and make the timeout longer.

Yes., if the test case work, this operation takes microseconds. But it might introduce flake in slower machines (CI workers).